docs/week-3.md

```diff
@@ -0,0 +1,568 @@
+
+```
+import React, { useState} from 'react';
+import LuckyNum from './LuckyNum';
+
+const Profile = () => (
+
+    <div>
+        <img src="profileImage.jpg"/>
+        <h1>Hi, my name is John! My favorite color is blue.</h1>
+        <LuckyNum />
+        <button>Likes: </button>
+    </div>
+);
+
+export default Profile;
+```
+
+Now in order to use `useState` we make a `const` array with a variable and a setter. We then set this equal
+to a call to `useState` and pass in the type of data we want to keep track of. For example, if we want to keep 
+track of someone's message, we would write this:
+
+```const [message, setMessage] = useState("Hello World")```
+
+We have initialized `message` to be "Hello World". If we want to change our message we would simply use 
+`setMessage`.
+
+```setMessage("This is our new message")```
+
+Now John can simply implement `useState` into his profile.
+
+```
+import React, { useState } from 'react';
+import LuckyNum from './LuckyNum';
+
+const Profile = () => (
+
+    // initialize the state to 0 because we start with 0 likes
+    const [profileLikes, setProfileLikes] = useState(0);
+
+    <div>
+        <img src="profileImage.jpg"/>
+        <h1>Hi, my name is John! My favorite color is blue.</h1>
+        <LuckyNum />
+        <button>Likes: {profileLikes}</button>
+    </div>
+);
+
+export default Profile;
+```
+
+Now all we have to do is create an `onClick` event to know when to increment the likes.
+
+```
+import React, { useState } from 'react';
+import LuckyNum from './LuckyNum';
+
+const Profile = () => (
+
+    // initialize the state to 0 because we start with 0 likes
+    const [profileLikes, setProfileLikes] = useState(0);
+
+    // handles incrementing the likes of John's profile
+    function handleLikes {
+        setProfileLikes(profileLikes + 1);
+    }    
+
+    <div>
+        <img src="profileImage.jpg"/>
+        <h1>Hi, my name is John! My favorite color is blue.</h1>
+        <LuckyNum />
+        <button onClick="handleLikes()">Likes: {profileLikes}</button>
+    </div>
+);
+
+export default Profile;
+```
+
+## Typing with Typescript
+As you may have noticed, react is normally implemented with Javascript. The only downside to using Javascript is 
+that we do not get type enforcement.
+
+### Why use Typescript in the first place?
+
+When developing an application with a lot of contributors, it is important to have some sort of structure to 
+your code. Typing (the act of giving our code types) allows us to not create errors and mess up a project 
+with other contributors on it. Typescript gives us the functionality of typing our components and data! 
+
+### How to type in React
+
+It is as easy as it seems. For example, lets type the `useState` that John used for likes on his profile and the
+`useState` for our message example:
+
+```
+const [profileLikes, setProfileLikes] = useState<Number>(0);
+const [message, setMessage] = useState<String>("Hello World")
+```
+
```

Comment:
Add a paragraph here (Or even better in its own section above this) that gives a description of the syntax for the types. Right now you're just showing examples for the function and not explaining how it could look for variables or other things we might add a type to